### PR TITLE
Track duration between token refreshes

### PR DESCRIPTION
**Pull request by vishalegbert-ttd**

### CODE REVIEW

File: src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java [hunk 1]
```diff
@@ -1027,15 +1027,37 @@ private void recordIdentityMapStats(RoutingContext rc, int inputCount) {
...
+            Gauge
+                    .builder("uid2.token_refresh_durations", durationSinceLastRefreshGauge, AtomicLong::get)
+                    .strongReference(true)
+                    .tag("api_contact", apiContact)
+                    .register(Metrics.globalRegistry);
+
+            return durationSinceLastRefreshGauge;
+        });
+
+        _refreshDurationMetrics.get(apiContact).set(durationSinceLastRefresh.getSeconds());
```
Comment (atarassov-ttd) [inline src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java:1058]: Doesn't `computeIfAbsent` return the entry? The `get()` should not be necessary?

File: src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java [hunk 2]
```diff
@@ -1027,15 +1027,37 @@ private void recordIdentityMapStats(RoutingContext rc, int inputCount) {
...
     }
 
+    private void recordRefreshDurationStats(RoutingContext rc, Duration durationSinceLastRefresh) {
+        String apiContact = getApiContact(rc);
+
+        _refreshDurationMetrics.computeIfAbsent(apiContact, k -> {
+            AtomicLong durationSinceLastRefreshGauge = new AtomicLong(0);
+            Gauge
+                    .builder("uid2.token_refresh_durations", durationSinceLastRefreshGauge, AtomicLong::get)
+                    .strongReference(true)
```
Comment (atarassov-ttd) [inline src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java:1051]: What is this for?

File: src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java [hunk 3]
```diff
@@ -1027,15 +1027,37 @@ private void recordIdentityMapStats(RoutingContext rc, int inputCount) {
...
             .register(Metrics.globalRegistry));
         ds.record(inputCount);
     }
 
+    private void recordRefreshDurationStats(RoutingContext rc, Duration durationSinceLastRefresh) {
+        String apiContact = getApiContact(rc);
+
+        _refreshDurationMetrics.computeIfAbsent(apiContact, k -> {
+            AtomicLong durationSinceLastRefreshGauge = new AtomicLong(0);
+            Gauge
```
Comment (atarassov-ttd) [inline src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java:1049]: Why is this a gauge? Wouldn't distribution summary be more appropriate here?

File: src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java [hunk 4]
```diff
@@ -1027,15 +1027,37 @@ private void recordIdentityMapStats(RoutingContext rc, int inputCount) {
...
+                    .builder("uid2.token_refresh_durations", durationSinceLastRefreshGauge, AtomicLong::get)
```
Comment (atarassov-ttd) [inline src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java:1050]: https://prometheus.io/docs/practices/naming/

> ...should have a suffix describing the unit, in plural form. Note that an accumulating count has total as a suffix, in addition to the unit if applicable.

File: src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java (hunk 2 above)
Comment (vishalegbert-ttd) [reply src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java]: Fixed

Comment (vishalegbert-ttd) [reply src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java]: Removed this

File: src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java (hunk 3 above)
Comment (vishalegbert-ttd) [reply src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java]: Wasn't familiar with distribution summary, I've changed it to use a DS instead

File: src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java (hunk 4 above)
Comment (vishalegbert-ttd) [reply src/main/java/com/uid2/operator/vertx/UIDOperatorVerticle.java]: Fixed

**Pull request merged**